keras_cv/models/segmentation/deeplab.py

```diff
@@ -56,136 +71,102 @@ class DeepLabV3(tf.keras.models.Model):
-                total_loss += l2_loss
-        self.optimizer.minimize(total_loss, self.trainable_variables, tape=tape)
-        return self.compute_metrics(images, y_true, y_pred, sample_weight=sample_weight)
-
-    def get_config(self):
-        config = {
-            "classes": self.classes,
-            "backbone": self._backbone_passed,
-            "decoder": self._decoder_passed,
-            "segmentation_head": self._segmentation_head_passed,
-        }
-        base_config = super().get_config()
-        return dict(list(base_config.items()) + list(config.items()))
+    inputs = utils.parse_model_inputs(input_shape, input_tensor)
+    height = input_shape[0]
+    width = input_shape[1]
+
+    x = inputs
+
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    if isinstance(backbone, str):
+        supported_premade_backbone = [
+            "resnet50_v2",
+        ]
+        if backbone not in supported_premade_backbone:
+            raise ValueError(
+                "Supported premade backbones are: "
+                f'{supported_premade_backbone}, received "{backbone}"'
+            )
+
+        if backbone == "resnet50_v2":
+            backbone = get_resnet_backbone(
+                backbone_weights, include_rescaling, **kwargs
+            )
+
+    else:
+        # TODO(scottzhu): Might need to do more assertion about the model
+        # What else do we want to test for? Shapes? This feels like too little, but
+        # more assertions feel like they'd be limiting.
+        if not isinstance(backbone, tf.keras.layers.Layer):
+            raise ValueError(
+                "Backbone need to be a `tf.keras.layers.Layer`, " f"received {backbone}"
+            )
+
+    feature_map = backbone(x)
+    output = SpatialPyramidPooling(dilation_rates=[6, 12, 18])(feature_map)
+    output = tf.keras.layers.UpSampling2D(
+        size=(height // feature_map.shape[1], width // feature_map.shape[2]),
+        interpolation="bilinear",
+    )(output)
+
+    if segmentation_head is None:
+        segmentation_head = get_segmentation_head(classes)
+
+    output = segmentation_head(output)
+    model = tf.keras.Model(inputs, output, name=name, **kwargs)
+
+    if backbone_weights is not None:
+        backbone.load_weights(backbone_weights)
+
+    return model
+
+
+def get_segmentation_head(classes):
+    return tf.keras.Sequential(
+        [
+            tf.keras.layers.Conv2D(
+                filters=256,
+                kernel_size=(1, 1),
+                padding="same",
+                use_bias=False,
+            ),
+            tf.keras.layers.BatchNormalization(),
+            tf.keras.layers.Activation("relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Conv2D(
+                filters=classes,
+                kernel_size=(1, 1),
+                padding="same",
+                use_bias=False,
+                activation="softmax",
```

Comment:
Is activation type hard coded? The loss function with `from_logits=True` might fail. 

cc. @qlzh727

Reply:
Good catch. I just noticed that we have a `SegmentationHead` in `__internal__`. Updating to use it, as it doesn't have an activation in the final layer. We can then pass the output of the head through an activation layer if it's requested in DeepLabV3.

Reply:
Updated!